torchtitan/experiments/deepseek_v3/model.py

```diff
@@ -525,32 +525,43 @@ def forward(self, hidden_states):
         # for each token, select top-k experts, and compute the weight for each expert
         topk_idx, topk_weight = self.gate(hidden_states)
         hidden_states = hidden_states.view(-1, hidden_states.shape[-1])
-        y = self.moe_infer(hidden_states, topk_idx, topk_weight).view(*orig_shape)
+        y = self.moe_forward(hidden_states, topk_idx, topk_weight).view(*orig_shape)
         if self.config.n_shared_experts is not None:
             y = y + self.shared_experts(identity)
         return y
 
-    @torch.no_grad()
-    def moe_infer(self, x, topk_ids, topk_weight):
+    def moe_forward(self, x, topk_ids, topk_weight):
```

Comment:
would be nice to add docstring to moe_forward (well, probably to other methods too) that document the shapes of the input tensors, e.g. `topk_ids: [seq_len, k]`